### PR TITLE
feat(HtmlWebpackPlugin): add metadata

### PR DESCRIPTION
**Pull request by yardenporat**

### CODE REVIEW

File: packages/scripts/test/application.unit.ts [hunk 1]
```diff
@@ -271,6 +271,22 @@ describe('Application', function () {
             expect(faviconHref).to.equal('favicon.ico');
         });
 
+        it('launches a built application with web environment and given metadata', async () => {
```
Comment (RomanYarik) [inline packages/scripts/test/application.unit.ts:274]: maybe repeat this test also for engineer, where we test the run command

**Review by RomanYarik — CHANGES_REQUESTED** [review]

File: packages/scripts/test/application.unit.ts (hunk 1 above)
Comment (yardenporat) [reply packages/scripts/test/application.unit.ts:274]: added

File: packages/engineer/src/feature/dev-server.feature.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ export interface DevServerConfig {
     publicPath?: string;
     title?: string;
     favicon?: string;
+    meta?: Record<string, string>;
```
Comment (AviVahl) [inline packages/engineer/src/feature/dev-server.feature.ts:18]: `meta`? `htmlMeta`?

File: packages/scripts/test/application.unit.ts [hunk 3]
```diff
@@ -271,6 +271,22 @@ describe('Application', function () {
...
+                meta: { unit_test: 'test_data' },
+            });
+            disposables.add(() => app.clean());
+
+            const { close, port } = await app.run();
+            disposables.add(close);
+
+            const page = await loadPage(`http://localhost:${port}/main.html`);
+            const testMetadata = await page.$('meta[name="unit_test"][content="test_data"]');
+            expect(testMetadata).to.not.be.null;
```
Comment (AviVahl) [inline packages/scripts/test/application.unit.ts:287]: to.not.equal(null)

File: packages/engineer/test/dev-server.unit.ts [hunk 4]
```diff
@@ -582,6 +585,21 @@ describe('engineer:dev-server', function () {
...
+            config: { port },
+        } = await setup({
+            basePath: engineConfigFixturePath,
+            featureName: 'engine-config/x',
+            meta: { unit_test: 'test_data' },
+        });
+
+        const page = await loadPage(`http://localhost:${port}/main.html`);
+
+        const testMetadata = await page.$('meta[name="unit_test"][content="test_data"]');
```
Comment (AviVahl) [inline packages/engineer/test/dev-server.unit.ts:600]: verify it's under head :P

File: packages/engineer/src/feature/gui.dev-server.env.ts [hunk 5]
```diff
@@ -87,6 +90,7 @@ function createDashboardConfig({
                 chunks: ['index'],
                 title,
                 favicon,
+                meta: htmlMeta ?? {},
```
Comment (AviVahl) [inline packages/engineer/src/feature/gui.dev-server.env.ts:93]: Why the fallback object?

Comment (yardenporat) [reply packages/engineer/src/feature/gui.dev-server.env.ts:93]: If it's empty, the default value is `{}`
but if htmlMeta is `undefined`, webpack doesn't know how to handle it
`'Html Webpack Plugin:\n  TypeError: Cannot convert undefined or null to object\n`
@AviVahl

**Pull request closed without merging**